### PR TITLE
Ft view single article 161875311

### PR DESCRIPTION
**Pull request by mindsworth**
#### What does this PR do?
implement view artcle functionality

#### Description of Task to be completed?
- viewing single article functionality implemented.
- modified the look and feel, and user experience across all pages
- installed `react-html-parser` to enable displaying content with html tags on the page.

#### How should this be manually tested?
Anyone with the right url can view any single article e.g `https://lotus-ah-frontend.herokuapp.com/viewarticle/This-is-the-first-article-by-femi-f343baa5-6e55-43c0-8411-1d9e0c611e2d`

#### What are the relevant pivotal tracker stories?
-[#161875311](https://www.pivotaltracker.com/story/show/161875311)

#### Screenshots (if appropriate)
screenshot 2018-11-16 at 1 27 05 pm

#### Questions:
N/A

### CODE REVIEW

File: src/actionTypes/article.js [hunk 1]
```diff
@@ -1,5 +1,27 @@
...
+
+// Fetch Article
+export const FETCH_ARTICLE = "FETCH_ARTICLE"
+export const FETCH_SINGLE_ARTICLE = "FETCH_SINGLE_ARTICLE"
+export const FETCH_ARTICLE_SUCCESS = "FETCH_ARTICLE_SUCCESS"
+export const FETCH_ARTICLE_FAILURE = "FETCH_ARTICLE_FAILURE"
+
+export const FETCH_SINGLE_ARTICLE_FAILURE = "FETCH_SINGLE_ARTICLE_FAILURE"
+export const FETCH_SINGLE_ARTICLE_SUCCESS = "FETCH_SINGLE_ARTICLE_SUCCESS"
+export const FETCH_SINGLE_ARTICLE_REQUEST = "FETCH_SINGLE_ARTICLE_REQUEST"
```
Comment (KaiserPhemi) [inline src/actionTypes/article.js]: #### Nitpick
Please add a new line here

**Review by KaiserPhemi — CHANGES_REQUESTED** [review]

File: src/styles/_signup.scss [hunk 2]
```diff
@@ -1,6 +1,6 @@
-h1 {
-    margin-bottom: 10%;
-}
+// h1 {
```
Comment (andela-phemmy) [inline src/styles/_signup.scss]: Good job, but you need to remove this commented code

File: src/containers/viewArticle/__test__/ViewArticle.test.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+// third-party libraries
+import { shallow, configure } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+
+// modules importation
+import ViewArticle from '../ViewArticle';
+import configureStore from '../../../store/configureStore';
+
+configure({ adapter: new Adapter() });
```
Comment (andela-phemmy) [inline src/containers/viewArticle/__test__/ViewArticle.test.js]: You don't need to configure enzyme here again, it has been configured in test-setup.js

File: src/components/signup/__test__/signupForm.test.js [hunk 4]
```diff
@@ -16,36 +16,29 @@ configure({ adapter: new Adapter() });
...
+ });
+
+ it('should render the signup page', () => {
+  const component = renderer.create(
+  <SignupPage register={ register } signup={{ registered: false } }/>,
+  { disableLifecycleMethods: true }
+  );
+  expect(component).toMatchSnapshot();
+ });
+
```
Comment (andela-phemmy) [inline src/components/signup/__test__/signupForm.test.js]: Try to  remove this blank line

File: src/components/signup/__test__/signupForm.test.js [hunk 5]
```diff
@@ -16,36 +16,29 @@ configure({ adapter: new Adapter() });
...
-  });
-
-  it('should render the update page', () => {
-    const component = renderer.create(
-      <UpdatePage updateAccount={{ }} location={{ verifiedUser }} updateUser= { updateUser }/>,
-      { disableLifecycleMethods: true }
-    );
-    expect(component).toMatchSnapshot();
-  });
+ const register = () => {
```
Comment (andela-phemmy) [inline src/components/signup/__test__/signupForm.test.js]: You can try to remove unused methods

File: src/components/reusables/userDetail/UserDetail.jsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
+// react libraries
+import React from 'react'
+
+// modules
```
Comment (andela-phemmy) [inline src/components/reusables/userDetail/UserDetail.jsx]: Try to change the import comment here to component

File: src/components/reusables/userAbout/_userAbout.scss [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    }
+  
+    .l-ah-user-info {
+      margin-left: 3rem;
+      flex-grow: 1;
+  
+      p {
+        margin: 0 0 2rem;
+      }
+
```
Comment (andela-phemmy) [inline src/components/reusables/userAbout/_userAbout.scss]: Try to remove this multiple blank lines

File: src/components/reusables/reaction/_reaction.scss [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    p {
+      font-weight: 600;
+      opacity: 1;
+      margin-top: .5rem;
+    }
+  }
+}
+
```
Comment (andela-phemmy) [inline src/components/reusables/reaction/_reaction.scss:49]: Try to remove this blank line

File: src/components/reusables/radialProgress/RadialProgress.jsx [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+		const strokeDasharray = 2 * radius * Math.PI;
+		
+		// Express progress as a percentage of stroke-dasharray. The difference between stroke-dasharray and this percentage is the stroke-dashoffset
+		const strokeDashoffset = progress > 0 ? (strokeDasharray - (strokeDasharray / 100 * progress)) : strokeDasharray;
+		
+		// We make a perfect circle, so it fits in a square.
+    const height = width;
+    
+    return (
+      <svg
```
Comment (andela-phemmy) [inline src/components/reusables/radialProgress/RadialProgress.jsx:24]: Try to reformat this svg

Comment (andela-phemmy) [reply src/components/reusables/radialProgress/RadialProgress.jsx:24]: To make the formating easy you can install prettier on your vscode

File: src/components/reusables/header/Header.jsx [hunk 10]
```diff
@@ -8,6 +8,7 @@ import PropTypes from 'prop-types';
 // components
 import HeaderSearch from '../headerSearch/HeaderSearch';
 import Dropdown from '../dropdown/Dropdown';
+import Logo from '../logo/Logo';
```
Comment (andela-phemmy) [inline src/components/reusables/header/Header.jsx]: Try to remove this blank line

File: src/components/reusables/comment/_comment.scss [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+          font-weight: 600;
+          opacity: 1;
+        }
+      }
+    }
+  }
+  .comment {
+    border: 1px solid $darker-gray;
+    position: relative;
+
```
Comment (andela-phemmy) [inline src/components/reusables/comment/_comment.scss]: Multiple blank line, try to remove

File: src/components/reusables/button/Button.jsx [hunk 12]
```diff
@@ -2,8 +2,7 @@
 import React, { Component } from 'react';
```
Comment (andela-phemmy) [inline src/components/reusables/button/Button.jsx]: You can try to remove this blank line as well

**Review by andela-phemmy — CHANGES_REQUESTED** [review]

File: src/actionTypes/article.js (hunk 1 above)
Comment (mindsworth) [reply src/actionTypes/article.js]: I have added a new line

File: src/actionTypes/article.js [hunk 13]
```diff
@@ -1,5 +1,27 @@
...
+export const UPDATE_ARTICLE_REQUEST = "UPDATE_ARTICLE_REQUEST"
+
+// Delete article
+export const DELETE_ARTICLE = "DELETE_ARTICLE"
+export const DELETE_ARTICLE_SUCCESS = "DELETE_ARTICLE_SUCCESS"
+export const DELETE_ARTICLE_FAILURE = "DELETE_ARTICLE_FAILURE"
+export const DELETE_ARTICLE_REQUEST = "DELETE_ARTICLE_REQUEST"
+
+// Fetch Article
+export const FETCH_ARTICLE = "FETCH_ARTICLE"
```
Comment (e0works) [inline src/actionTypes/article.js:20]: Is there any reason we have multiple fetch single article action types?

File: src/containers/viewArticle/__test__/ViewArticle.test.js (hunk 3 above)
Comment (mindsworth) [reply src/containers/viewArticle/__test__/ViewArticle.test.js]: Please ... I don't get you here.

File: src/styles/_signup.scss (hunk 2 above)
Comment (mindsworth) [reply src/styles/_signup.scss]: done

File: src/components/signup/__test__/signupForm.test.js (hunk 4 above)
Comment (mindsworth) [reply src/components/signup/__test__/signupForm.test.js]: done

File: src/components/signup/__test__/signupForm.test.js (hunk 5 above)
Comment (mindsworth) [reply src/components/signup/__test__/signupForm.test.js]: done

File: src/components/reusables/userDetail/UserDetail.jsx (hunk 6 above)
Comment (mindsworth) [reply src/components/reusables/userDetail/UserDetail.jsx]: done

File: src/components/reusables/userAbout/_userAbout.scss (hunk 7 above)
Comment (mindsworth) [reply src/components/reusables/userAbout/_userAbout.scss]: done

File: src/components/reusables/reaction/_reaction.scss (hunk 8 above)
Comment (mindsworth) [reply src/components/reusables/reaction/_reaction.scss:49]: done

File: src/components/reusables/reaction/Reaction.jsx [hunk 14]
```diff
@@ -0,0 +1,82 @@
+import React, { Component } from 'react'
```
Comment (e0works) [inline src/components/reusables/reaction/Reaction.jsx:2]: Please comment on the imports and also move the `export default` to last line

File: src/containers/viewArticle/__test__/ViewArticle.test.js (hunk 3 above)
Comment (andela-phemmy) [reply src/containers/viewArticle/__test__/ViewArticle.test.js]: Check out this article https://stackoverflow.com/questions/46627353/where-should-the-enzyme-setup-file-be-written

File: src/components/reusables/userAbout/UserAbout.jsx [hunk 15]
```diff
@@ -0,0 +1,27 @@
+import React from 'react'
```
Comment (e0works) [inline src/components/reusables/userAbout/UserAbout.jsx:2]: Please comment on the imports here and also move the `export default` after the class block

File: src/components/reusables/radialProgress/RadialProgress.jsx (hunk 9 above)
Comment (mindsworth) [reply src/components/reusables/radialProgress/RadialProgress.jsx:24]: I have prettier installed... A sample of the right formatting will help.

File: src/components/reusables/header/Header.jsx (hunk 10 above)
Comment (mindsworth) [reply src/components/reusables/header/Header.jsx]: done

File: src/components/reusables/comment/_comment.scss (hunk 11 above)
Comment (mindsworth) [reply src/components/reusables/comment/_comment.scss]: done

File: src/actionTypes/article.js (hunk 13 above)
Comment (mindsworth) [reply src/actionTypes/article.js:20]: done

File: src/components/reusables/reaction/Reaction.jsx (hunk 14 above)
Comment (mindsworth) [reply src/components/reusables/reaction/Reaction.jsx:2]: done

File: src/components/reusables/userAbout/UserAbout.jsx (hunk 15 above)
Comment (mindsworth) [reply src/components/reusables/userAbout/UserAbout.jsx:2]: done

Comment (mindsworth) [reply src/components/reusables/userAbout/UserAbout.jsx:2]: done

**Review by andela-phemmy — APPROVED** [review]

File: src/containers/viewArticle/__test__/ViewArticle.test.js (hunk 3 above)
Comment (KaiserPhemi) [reply src/containers/viewArticle/__test__/ViewArticle.test.js]: @mindsworth 
How far with this?

File: src/components/signup/__test__/signupForm.test.js [hunk 16]
```diff
@@ -2,49 +2,56 @@
 import React from 'react';
 
 // third-party libraries
-import { shallow, configure } from 'enzyme';
+import {
+  shallow,
+  configure
+} from 'enzyme';
 import Adapter from 'enzyme-adapter-react-16';
```
Comment (KaiserPhemi) [inline src/components/signup/__test__/signupForm.test.js:9]: #### Critical
I thought enzyme was configured somewhere in a `testSetup` file ? 
Why configure enzyme in every test file?

**Review by KaiserPhemi — CHANGES_REQUESTED** [review]

File: src/components/signup/__test__/signupForm.test.js (hunk 16 above)
Comment (mindsworth) [reply src/components/signup/__test__/signupForm.test.js:9]: Fixed

**Review by KaiserPhemi — APPROVED** [review]

**Pull request merged**